bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/dto/Characteristic.java

```diff
@@ -0,0 +1,978 @@
+import org.openhab.binding.homekit.internal.enums.CharacteristicType;
+import org.openhab.binding.homekit.internal.enums.DataFormatType;
+import org.openhab.binding.homekit.internal.enums.StatusCode;
+import org.openhab.binding.homekit.internal.persistence.HomekitTypeProvider;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.library.CoreItemFactory;
+import org.openhab.core.semantics.SemanticTag;
+import org.openhab.core.semantics.model.DefaultSemanticTags.Point;
+import org.openhab.core.semantics.model.DefaultSemanticTags.Property;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.type.ChannelDefinitionBuilder;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.thing.type.ChannelTypeBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.thing.type.StateChannelTypeBuilder;
+import org.openhab.core.types.StateDescriptionFragment;
+import org.openhab.core.types.StateDescriptionFragmentBuilder;
+import org.openhab.core.types.StateOption;
+import org.osgi.framework.Bundle;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * HomeKit characteristic DTO.
+ * Used to deserialize individual characteristics from the /accessories endpoint of a HomeKit bridge.
+ * Each characteristic has a type, instance ID (iid), value, permissions (perms), and format.
+ * This class also includes a method to convert the characteristic to an openHAB ChannelType, if possible.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class Characteristic {
+    public @NonNullByDefault({}) String type; // 25 = public.hap.characteristic.on
+    public @NonNullByDefault({}) String format; // e.g. "bool"
+    public @NonNullByDefault({}) List<String> perms; // e.g. ["pr", "pw", "ev"]
+    public @NonNullByDefault({}) Long iid; // e.g. 10
+    public @NonNullByDefault({}) String unit; // e.g. "celsius" or "percentage"
+    public @NonNullByDefault({}) Double maxValue; // e.g. 100
+    public @NonNullByDefault({}) Double minValue; // e.g. 0
+    public @NonNullByDefault({}) Double minStep;
+    public @NonNullByDefault({}) JsonElement value; // e.g. true, 23, "Some String"
+    public @NonNullByDefault({}) String description;
+    public @NonNullByDefault({}) Boolean ev; // e.g. true (events requested)
+    public @NonNullByDefault({}) Long aid; // e.g. 10
+    public @NonNullByDefault({}) @SerializedName("valid-values") List<Integer> validValues;
+    public @NonNullByDefault({}) @SerializedName("valid-values-range") List<Integer> validValuesRange;
+    public @NonNullByDefault({}) Integer status;
+
+    /**
+     * Returns the {@link Content} for this characteristic. Some Characteristics have variable values and others remain
+     * static over time. The latter return a 'Property' record and the latter return a 'ChannelDefinition' record.
```

Comment:
I spotted "latter" two times and resorted to AI for double-checking, and it came up with this suggestion:
```suggestion
     * Returns the {@link Content} for this characteristic. Some characteristics have values that may change over time,
     * while others remain static. Characteristics with static values return a {@code Property} record,
     * whereas characteristics with dynamic values return a {@code ChannelDefinition} record.
```

(otherwise I think you'd just need to replace the latter "latter" with "former" 😉)